internal/ihop/builder.go

```diff
@@ -0,0 +1,84 @@
+package ihop
+
+//go:generate faux --interface ImageScanner --output fakes/image_scanner.go
+type ImageScanner interface {
+	Scan(tag string) (SBOM, error)
+}
+
+//go:generate faux --interface ImageBuildPromise --output fakes/image_build_promise.go
+type ImageBuildPromise interface {
+	Resolve() (Image, SBOM, error)
+}
+
+type builderInput struct {
+	def      DefinitionImage
+	platform string
+	results  chan<- builderResult
+}
+
+type builderResult struct {
+	image Image
+	sbom  SBOM
+	err   error
+}
+
+type Builder struct {
+	client  ImageClient
+	scanner ImageScanner
+
+	inputs chan builderInput
+}
+
+func NewBuilder(client ImageClient, scanner ImageScanner, workerCount int) Builder {
```

Comment:
The builder process is pretty complicated from the optimizations you've made.  I might benefit from a short comment at the top of this file describing how this works for future reference.

Reply:
I've added a bunch of inline docs in the latest update.

Reply:
TY :)